test/battle/move_effect/entrainment.c

```diff
@@ -14,6 +14,50 @@ AI_DOUBLE_BATTLE_TEST("AI prefers Entrainment'ing good abilities onto partner wi
     }
 }
 
-TO_DO_BATTLE_TEST("Entrainment changes the target's Ability to match the user's");
-TO_DO_BATTLE_TEST("Entrainment fails if the user's ability has cantBeCopied flag");
-TO_DO_BATTLE_TEST("Entrainment fails if the targets's ability has cantBeOverwritten flag");
+SINGLE_BATTLE_TEST("Entrainment changes the target's Ability to match the user's")
+{
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET) { Ability(ABILITY_TELEPATHY); }
+        OPPONENT(SPECIES_WOBBUFFET) { Ability(ABILITY_SHADOW_TAG); }
+    } WHEN {
+        TURN { MOVE(player, MOVE_ENTRAINMENT); }
+    } THEN {
+        EXPECT(opponent->ability == ABILITY_TELEPATHY);
+    }
+}
+
+SINGLE_BATTLE_TEST("Entrainment fails if the user's ability has cantBeCopied flag")
+{
+    ASSUME(gAbilitiesInfo[ABILITY_MULTITYPE].cantBeCopied);
+
+    GIVEN {
+        PLAYER(SPECIES_ARCEUS)   { Ability(ABILITY_MULTITYPE); }
+        OPPONENT(SPECIES_WOBBUFFET) { Ability(ABILITY_SHADOW_TAG); }
+    } WHEN {
+        TURN { MOVE(player, MOVE_ENTRAINMENT); }
+    } SCENE {
+        MESSAGE("But it failed!");
+    } THEN {
+        EXPECT(player->ability == ABILITY_MULTITYPE);
+        EXPECT(opponent->ability == ABILITY_SHADOW_TAG);
+    }
+}
+
+SINGLE_BATTLE_TEST("Entrainment fails if the target's ability has cantBeOverwritten flag")
+{
+    ASSUME(gAbilitiesInfo[ABILITY_MULTITYPE].cantBeOverwritten);
+
+    GIVEN {
```

Comment:
```suggestion
    GIVEN {
        ASSUME(gAbilitiesInfo[ABILITY_MULTITYPE].cantBeOverwritten);
```
`ASSUME` outside of `GIVEN`